File: PiDay/Core/Domain/IndexingConvention.swift
import Foundation

// WHY: How we label digit positions is a user-facing preference, not a data concern.
// Keeping it in Domain makes it available to both ViewModel and View layers.
enum IndexingConvention: String, CaseIterable, Identifiable {
    case oneBased
    case zeroBased

    var id: String { rawValue }

    var label: String {
        switch self {
        case .oneBased:
            return "1-based"
        case .zeroBased:
            return "0-based"
        }
    }

    var explainer: String {
        switch self {
        case .oneBased:
            return "Digit 1 is the first digit after the decimal point."
        case .zeroBased:
            return "Digit 0 is the first digit after the decimal point."
        }
    }

    func displayPosition(for storedPosition: Int) -> Int {
        switch self {
        case .oneBased:
            return storedPosition
        case .zeroBased:
            return storedPosition - 1
        }
    }
}

File: PiDayAndroid/app/src/main/java/academy/glasscode/piday/core/domain/IndexingConvention.kt
package academy.glasscode.piday.core.domain

enum class IndexingConvention {
    ONE_BASED, ZERO_BASED;

    val label: String get() = when (this) {
        ONE_BASED  -> "1-based"
        ZERO_BASED -> "0-based"
    }

    val explainer: String get() = when (this) {
        ONE_BASED  -> "Digit 1 is the first digit after the decimal point."
        ZERO_BASED -> "Digit 0 is the first digit after the decimal point."
    }

    fun displayPosition(storedPosition: Int): Int = when (this) {
        ONE_BASED  -> storedPosition
        ZERO_BASED -> storedPosition - 1
    }
}
